app/AbstractRequest.php

```diff
@@ -323,12 +323,15 @@ protected function writeResponseToCache(): void
         Cache::tags($this->cacheTags)->put(
             $this->cacheKey(),
             [
-                // matches the argument order for Response constructor
-                $this->response->getStatusCode(),
-                $this->response->getHeaders(),
-                (string) $this->response->getBody(),
-                $this->response->getProtocolVersion(),
-                $this->response->getReasonPhrase(),
+                'logs' => $this->sentLogs,
```

Comment:
This is the big idea, the cached object contains both the array of log file names, *and* enough data to rehydrate a Guzzle Response object.